app/controllers/payments_controller.rb

```diff
@@ -0,0 +1,30 @@
+class PaymentsController < ApplicationController
+  before_action :authenticate_user!
+  def index
+    @category = Category.find(params[:category_id])
+    @payments = Payment.where(author: current_user, category_id: @category.id).order(updated_at: :desc)
+    @total = Payment.where(author: current_user, category_id: @category.id).sum(:amount)
+  end
+
+  def new
+    @payment = Payment.new
+  end
+
+  def create
+    @payment = Payment.new(payment_params)
+    @payment.author_id = current_user.id
+    category = Category.find(params[:category_id])
+    @payment.category_id = category.id
+    if @payment.save
+      redirect_to category_payments_path
```

Comment:
[OPTIONAL ] Adding notices to your sucess and fail paths will give users a better user experience.
**Example**
```
def create
  @record = Record.new(record_params)
  if @record.save
    redirect_to @record, notice: 'Record was successfully created.'
  else
    render :new
  end
end
```